src/sbvr-api/permissions.ts

```diff
@@ -684,6 +684,161 @@ const getAlias = (name: string) => {
+		}
+
+		const originalAbstractSQLModel = sbvrUtils.getAbstractSqlModel({
+			vocabulary,
+		});
+
+		const rewrite = (object: AnyObject) => {
+			if (_.has(object, '$') && _.isArray(object['$'])) {
+				const arr = object['$'];
+				if (
+					arr.length === 2 &&
+					_.isArray(arr[1]) &&
+					arr[1].length === 2 &&
+					_.isString(arr[1][0])
+				) {
+					// Skip this if we already shortcut this connection
+					if (arr[1][0].endsWith('$bypass')) {
+						return;
+					}
+
+					const targetResource = abstractSqlModel.tables[arr[1][0]];
+					let targetResourceEscaped = sqlNameToODataName(arr[1][0]);
+					if (targetResource) {
+						targetResourceEscaped = sqlNameToODataName(targetResource.name);
+					}
+
+					// This is either a transalted or bypassed resource we don't
+					// mess with these
+					if (targetResourceEscaped.indexOf('$') > 0) {
+						return;
+					}
+
+					let foundCanAccessLink = false;
+
+					try {
+						const odata = memoizedParseOdata(`/${targetResourceEscaped}`);
+
+						const collapsedPermissionFilters = buildODataPermission(
+							permissionsLookup,
+							methodPermissions.GET,
+							vocabulary,
+							targetResourceEscaped,
+							odata,
+						);
+
+						_.set(
+							odata,
+							['tree', 'options', '$filter'],
+							collapsedPermissionFilters,
+						);
+
+						const canAccessFunction: ResourceFunction = function(
+							property: AnyObject,
+						) {
+							// remove method property so that we won't loop back here again at this point
+							delete property.method;
+
+							if (!this.defaultResource) {
+								throw new Error(`No resource selected in AST.`);
+							}
+
+							const targetResourceAST = this.NavigateResources(
+								this.defaultResource,
+								property.name,
+							);
+
+							const targetResourceName = sqlNameToODataName(
+								targetResourceAST.resource.name,
+							);
+							const currentResourceName = sqlNameToODataName(
+								this.defaultResource.name,
+							);
+
+							if (
+								currentResourceName === targetResourceEscaped &&
+								targetResourceName === escapedName
+							) {
+								foundCanAccessLink = true;
+							}
+							// return a true expression to not select the relationship, which might be virtual
+							// this should be a boolean expression, but needs to be a subquery in case it
+							// is wrapped in an `or` or `and`
+							return [
+								'Exists',
+								['SelectQuery', ['Select', [['Alias', ['Number', 1], 'id']]]],
+							];
+						};
+
+						// We need execute the
+						const odata2AbstractSQL = new OData2AbstractSQL(
+							originalAbstractSQLModel,
+							{
+								canAccess: canAccessFunction,
+							},
+						);
+
+						odata2AbstractSQL.match(odata.tree, 'GET', [], odata.binds.length);
+
+						if (foundCanAccessLink) {
+							arr[1][0] = `${arr[1][0]}$bypass`;
+						}
+					} catch (e) {
+						if (e === constrainedPermissionError) {
+							// ignore
+							return;
+						}
+					}
```

Comment:
This currently ignores all errors, I expect you want:
```suggestion
					} catch (e) {
						if (e === constrainedPermissionError) {
							// ignore
							return;
						}
						throw e;
					}
```